src/emitters/emitter.sqns.ts

```diff
@@ -997,4 +999,19 @@ export class SqnsEmitter implements IEmitter {
     const urlParts = queueUrl.split('/')
     return urlParts[urlParts.length - 1]
   }
+
+  private saveApproximateReceiveCount(receivedMessage: Message, message: IMessage<any>) {
+    const approximateRecievedCount = this.getApproximateReceiveCount(receivedMessage);
+    if (message.messageAttributes) {
```

Comment:
if messageAttributes do not exist, we should create them?

Reply:
Got it, done, please check.